blockchain/blockindex.go

```diff
@@ -260,6 +260,16 @@ func (node *blockNode) RelativeAncestorCtx(distance int32) HeaderCtx {
 	return ancestor
 }
 
+// IsAncestor returns if the other node is an ancestor of this block node.
+func (node *blockNode) IsAncestor(otherNode *blockNode) bool {
+	ancestor := node.Ancestor(otherNode.height)
+	if ancestor == node {
```

Comment:
Both the arg, the method, and the method target are pointers, so this will only succeed if all the pointers across the structure are consistent. Instead, I think we should refer here so we do an actual equality check. We may actually want to consider making an `Equals` method for `blockNode`.

Reply:
Yeah I'll add an Equals here

Reply:
Addressed in the latest push